IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -0,0 +1,497 @@
+        if (error)
+        {
+            *error = MSIDCreateError(
+                MSIDKeychainErrorDomain, (NSInteger)status, errorMessage, nil, nil, nil, context.correlationId, nil);
+        }
+        return nil;
+    }
+    
+    NSArray *items = CFBridgingRelease(cfItems);
+
+    query = [defaultQuery mutableCopy];
+    // Note: For efficiency, use kSecUseItemList to query the items returned above rather actually querying
+    // the keychain again. With this second query we can set a specific kSecMatchLimit which lets us get the data
+    // objects.
+    query[(id)kSecUseItemList] = items;
+    query[(id)kSecMatchLimit] = @(items.count + 1); // always set a limit > 1 so we consistently get an NSArray result
+    query[(id)kSecReturnAttributes] = @YES;
+    query[(id)kSecReturnData] = @YES;
+
+    CFTypeRef cfItemDicts = nil;
+    status = SecItemCopyMatching((CFDictionaryRef)query, &cfItemDicts);
+    NSArray *itemDicts = CFBridgingRelease(cfItemDicts);
+
+    NSMutableArray<MSIDAccountCacheItem *> *accountList = [NSMutableArray new];
+    for (NSDictionary *dict in itemDicts)
+    {
+        NSData *jsonData = dict[(id)kSecValueData];
+        if (jsonData)
+        {
+            MSIDAccountCacheItem *account = (MSIDAccountCacheItem *)[serializer deserializeAccountCacheItem:jsonData];
+            if (account == nil)
+            {
+                MSID_LOG_WARN(context, @"Failed to deserialize account");
+                continue;
+            }
+            [accountList addObject:account];
+        }
+    }
+    return accountList;
+}
+
+// Remove one or more accounts from the keychain that match the key.
+- (BOOL)removeItemsWithAccountKey:(MSIDCacheKey *)key
+                          context:(id<MSIDRequestContext>)context
+                            error:(NSError **)error
+{
+    MSID_TRACE;
+    MSID_LOG_INFO( context, @"Remove keychain items, key info (account: %@ service: %@)", _PII_NULLIFY(key.account), _PII_NULLIFY(key.service));
+    MSID_LOG_INFO_PII(context, @"Remove keychain items, key info (account: %@ service: %@)", key.account, key.service);
+
+    NSMutableDictionary *query = [[self defaultAccountQuery:key] mutableCopy];
+    query[(id)kSecMatchLimit] = (id)kSecMatchLimitAll;
+    query[(id)kSecReturnAttributes] = @YES;
+
+    MSID_LOG_INFO(context, @"Trying to delete keychain items...");
+    OSStatus status = SecItemDelete((CFDictionaryRef)query);
+    MSID_LOG_INFO(context, @"Keychain delete status: %d", (int)status);
+
+    if (status != errSecSuccess && status != errSecItemNotFound)
+    {
+        NSString *errorMessage = @"Failed to remove multiple accounts from keychain";
+        MSID_LOG_WARN(context, @"%@ (%d)", errorMessage, status);
+        if (error)
+        {
+            *error = MSIDCreateError(
+                MSIDKeychainErrorDomain, (NSInteger)status, errorMessage, nil, nil, nil, context.correlationId, nil);
+        }
+        return FALSE;
+    }
+
+    return (status == errSecSuccess);
```

Comment:
it returns false if keychain is empty. consistent with ios?

Reply:
updated to return TRUE like iOS